src/App.vue

```diff
@@ -0,0 +1,253 @@
+<script setup>
+import { ref, onMounted } from 'vue'
+
+const origins = ref([])
+const inputText = ref('')
+const placeholderURL = ref('https://hoppscotch.io')
+const errorMessage = ref('')
+
+const DEFAULT_ORIGIN_LIST = ["https://example.com", "https://another-example.com"]
+
+const getOriginList = () =>
+  new Promise((resolve) => {
+    chrome.storage.sync.get(["originList"], async (items) => {
+      if (!items || !items.originList) {
+        await storeOriginList(DEFAULT_ORIGIN_LIST)
+        resolve(DEFAULT_ORIGIN_LIST)
+      } else {
+        resolve(JSON.parse(items.originList))
+      }
+    })
+  })
+
+const storeOriginList = (originList) =>
```

Comment:
add type for `originList`.

```suggestion
const storeOriginList = (originList: string[) =>
```

 i'm not adding seperate comments for all the typescript errors. please resolve all the typescript issues. they should pop up in your editor after setting `lang="ts"` in script setup.

Reply:
Resolved, added types to all errors and checked the code, it is working fine, will raise a commit on this PR soon.